server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java

```diff
@@ -1211,6 +1219,72 @@ void oauthShouldFailWhenIntrospectTokenServerError() throws Exception {
             client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
             assertThat(client.getReplyString()).contains("NO AUTHENTICATE processing failed.");
         }
+
+        @Test
+        void oauthShouldImpersonateFailWhenNOTDelegated() throws Exception {
+            String oauthBearer = OIDCSASLHelper.generateOauthBearer(USER2.asString(), OidcTokenFixture.VALID_TOKEN);
+            IMAPSClient client = imapsClient(port);
+            client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
+            assertThat(client.getReplyString()).contains("NO AUTHENTICATE Delegation is forbidden");
+        }
+
+        @Test
+        void oauthShouldImpersonateSuccessWhenDelegated() throws Exception {
+            createImapServerForImpersonateTest();
+
+            String oauthBearer = OIDCSASLHelper.generateOauthBearer(USER2.asString(), OidcTokenFixture.VALID_TOKEN);
+            IMAPSClient client = imapsClient(port);
+            client.sendCommand("AUTHENTICATE OAUTHBEARER " + oauthBearer);
+            assertThat(client.getReplyString()).contains("OK AUTHENTICATE completed.");
```

Comment:
Can we create a specific mailbox in user 2 account ? Then listing it after auth would ensure that we are well connected as USER2 and not as USER.

Reply:
It is already at `impersonationShouldWorkWhenDelegated`